Accessibility

Hombre2014 opened the pull request:
The changes have been made according to the accessibility checklist as follow:
- Page title - OK
- All images have alt text attribute
- Text heading the structure is good.
- Color contrast. Many changes have been made to achieve a good contrast. By doing this colors that were introduced do not correspond to Figma design, (where changes have been made).
- Resize the design looks good on large and very text sizes with no overflow.
- Interaction - all links, menu items, and forms are accessible via a keyboard
- The basic structure of the page is checked and it looks good.

Hombre2014 commented:
This is a pair-programming stage with @kakarrot92 .

alonsofl commented on index.html:
```diff
@@ -176,7 +176,7 @@ <h2 class="work-title hidden-on-small left">Uber Navigation</h2>
       <div class="big-about">
         <div class="left-part">
           <h2 class="title myself">About<br>Myself</h2>
```
- [ ] You have done a great job making your portfolio more accessible but here is an issue with "Text headings", remember that heading (`<hX>`) levels should have a meaningful hierarchy and you are using an `<h2>` as a section title, but you are also using the same `<h2>` tag for your project titles. I think it could be a good idea to keep the `<h2>` tag for the sections titles and give a "less" important tag (semantically speaking) to the project titles :wink:

  ![image](https://user-images.githubusercontent.com/81522435/139878695-4c145348-5ac6-4eed-8cb9-d4e8ba809734.png)

Hombre2014 replied:
Hi, @alonsofl! I think this is a kind of misunderstanding of the concept I am using in the design. On the lines 147-148 I have:
`<h2 class="work-title hidden-on-big">Multi-Post Stories</h2>`
`<h2 class="work-title hidden-on-small left">Uber Navigation</h2>`
These are not nested <hx>'s. These are one and the same `<h2>` headings shown on different screen sizes. There is nothing semantically wrong here. When you compare it to the next section "About" on line 178, it is again `<h2>` because it is at the same level, but in a different section. The "Uber Navigation" is part of the "Works" section and has only `<h2>` levels of headings. The "About" is a different section on the same level as the "Works" section, so the heading will remain the same. Please, let me know if you disagree. I am sending a snapshot from the Dev Tools plugin report on the structure of my website:

![image](https://user-images.githubusercontent.com/8495547/139919680-18b5e798-30a3-4090-b4aa-81ccfdcd78f1.png)

I have only `<h1>` and `<h2>` tags in the HTML file, but some of them are shown on a small screen, some of them shown on a big screen, as per Figma design. Please, have a look:

https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=34%3A848

So, I will not change the headings.

alonsofl replied:
Hello @Hombre2014, I get your point about the headings and I noticed that you use different titles for different breakpoints. My suggestion is because headings `<hX>` give different "importance" to a title (semantically speaking). That's why I recommend using the `<h2>` tag for the Section title, which I think is more important than each project title (consider those as title levels). For example, if you see the other templates, these have a project section title, so this could be the same case and each project title would use the `<h3>` tag because these are "inside" or have a "lower-level" of importance. Once again, headings are not for styling/text size (for that we could use CSS), headings represent the importance of a title. I hope this explanation is a bit clearer :raised_hands:

Hombre2014 replied:
Hi @alonsofl. I also see your point and when I looked at the other templates, yes, there are "My recent works" or "My projects" titles and then a list of the recent project. In this case, of course, it is semantically correct to use the `<h2>` title for the Projects and the `<h3>` titles for each project itself. This is valid for all of templates 2, 3, 4 and 5. But there is no such title or heading on template 1! Please, have a look. That is why I do not have h2 for My Recent Projects because I do not have a section title My recent project. I only have 4 projects listed! Template number one is without this type of title. If you want it, please, correct the template!

alonsofl replied:
@ Hombre2014, I gave that as an example, I am aware that your template does not have a title for the Projects section. But with that premise, isn't the "About Section title" more important than each project title? Also, is this "About Section title" less important than your headline title? Then why would you use the `<h2>` for a section title and also for a less important title? Or do you consider that the "About section title" is equally as important as each project?

Hombre2014 replied:
@alonsofl Well, we can continue arguing till the end of ... I see your point that About is the same level as My Recent Works and they deserve to have an h2 titles, but the template has not My Recent works section title, as I already said. I either have to introduce a new h2 title My Recent works or Projects as in all other templates and then each project will have an h3 title, but by doing this I will not follow the template on Figma. So, please, either fix the template or accept my point of view. On Figma they specify the requirements as follow:

Header/2:

![image](https://user-images.githubusercontent.com/8495547/139945658-d805fcb4-c611-43cc-8cfb-92ca79c55721.png)

Header/2:

![image](https://user-images.githubusercontent.com/8495547/139945734-487df415-a9b6-465a-b71e-dcd08f6f2958.png)

alonsofl commented on index.html:
```diff
@@ -19,7 +19,7 @@
+  <title>Yuriy's Mircroverse Portfolio</title>
 </head>
 
 <body>
```
- [ ] There is a contrast issue with your navigation links. I think it could be a good idea to update the hover colors to make your page more accessible :hugs:

  Your design:
  ![image](https://user-images.githubusercontent.com/81522435/139879623-6a77f1e7-1ebe-41dd-a377-0f09a9b71a75.png)

Hombre2014 replied:
I see the picture you attached and it is very bad. I do have a transition on the navigation links and the transition is on the color and background color. So, the transition, I agree is a little bit long - 2 seconds. For the part of the transition, yeah, it might go through such a bad contrast, but once the transition is finished it will be OK. I am really surprised how you were able to capture this screenshot because if you waited for two seconds you will get this:

![image](https://user-images.githubusercontent.com/8495547/139920939-2dadb690-4324-4ff1-b0a1-5df73cd97841.png)

I will shorten the time of the transition, so the visitor will not have this experience, or it will have it for a really short time, so it will not be noticeable.

D4nielJ replied:
- [ ] Good job making your animation shorter. This is part of thinking in everyone. Unfortunately, the contrast-ratio is still to low ("2.88"), we can increase this to four by changing it to this shade (#6070ff) or another shade that you find convenient. As long as it gets higher than four.

![image](https://user-images.githubusercontent.com/79114233/139925924-67ec366d-f8b5-4a19-8386-86ad54d83f0a.png)

Hombre2014 replied:
Thanks for noticing it. Yes, these are the original colors stated in the Figma template and I used them as I had to pass my mobile part of the portfolio project. So, now I assume, I have to change them because they are not looking good as the accessibility required. OK. I made the change and have got a 4.57:1 contrast ratio, which Chrome DevTools says is so OK (green).

alonsofl reviewed (changes requested):
Hi @Hombre2014 and @kakarrot92,

You have done a great job so far! :muscle:
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## To highlight :dart:
- Descriptive title :heavy_check_mark:
- Great use of the `alt` attribute :heavy_check_mark:
- Keyboard access :heavy_check_mark:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Hombre2014 commented:
Please, see my comments above. I shortened the transition time for the hover event over the navigation links. The headings I think are OK semantically.

D4nielJ commented on index.html:
```diff
@@ -244,22 +248,22 @@ <h2 class="title myself">About<br>Myself</h2>
                   required>
               </label>
-              <label for="msg">
+              <label id="last" class="label-text" for="msg">Please write your message<span class="red">*</span>
```
[OPTIONAL] Good job adding labels to your inputs so they're accessible to everyone. I think you could hide these to the average user and make them accessible only to screen readers. [Read more about it here.](https://css-tricks.com/inclusively-hidden/) It will help you match the design in Figma while also keeping your web-accessible.

Hombre2014 replied:
I see your point. If I remove the labels it will not pass the Dev Tools toolbox test, because it will say there are empty labels. That is why I put them. I am really confused and frustrated because there is one template and requirements and now you are asking me to make something else, and either way, I will be punishable. If I make it as the Figma template without labels, somebody could reject my pull request, because one of the requirements of the accessibility step is not fulfilled (interaction: https://www.w3.org/WAI/test-evaluate/preliminary/#interaction). So, when I made this part to fulfill the step now I am against the Figma design, which does not require labels. I will not introduce this change, since it is optional and I need more clarity about what to follow and what is more important.

D4nielJ replied:
Hello, I'm sorry my explanation wasn't clear enough. I wasn't asking you to remove the labels from your HTML document. I was asking you to hide them from plain sight through a class in CSS. This is usually a class call `.screen-readers-only`.

The screen readers will read them since they're present in the HTML file, but for users not using screen readers they won't be present and the design will remain intact.

That's why the name of the article is "Inclusively Hidden". Also, I marked this as optional and optional requirements won't stop a project from being approved.

I hope you find this insightful and you can get my suggestion. If you have any doubts, don't hesitate to reach to me on Slack.

Hombre2014 replied:
@D4nielJ Thanks for your clarification. Today we are starting with Bootstrap and it will be a good opportunity to implement this class in my future design.

D4nielJ replied:
In Bootstrap they have implementations ready to go. So you don't even need to add them to your stylesheets. [Check them out](https://getbootstrap.com/docs/5.1/helpers/visually-hidden/). 😉

D4nielJ reviewed (changes requested):
## Hello @Hombre2014, 

Great work on making the changes requested by the previous reviewer(s) 👏

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Hombre2014 commented:
Please, take a look at my comments above. I addressed all the issues previous reviewer requested. The changes have been implemented and all the issues, that are required from the two reviews have been resolved.

carlos-ssh reviewed (approved):
# 🍾🍾 **STATUS APPROVED** 👏🏽👏🏽

Hi @Hombre2014 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



![Approved](https://user-images.githubusercontent.com/22258274/138466108-6b14c541-6a88-44c5-ba76-c64992ac1835.gif)


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
